tests/ResponseTest.php

```diff
@@ -9,167 +9,223 @@ class ResponseTest extends TestCase
 {
     public function testResponseShouldBeChunkedByDefault()
     {
-        $expected = '';
-        $expected .= "HTTP/1.1 200 OK\r\n";
-        $expected .= "X-Powered-By: React/alpha\r\n";
-        $expected .= "Transfer-Encoding: chunked\r\n";
-        $expected .= "Connection: close\r\n";
-        $expected .= "\r\n";
-
+        $buffer = '';
         $conn = $this
             ->getMockBuilder('React\Socket\ConnectionInterface')
             ->getMock();
         $conn
             ->expects($this->once())
             ->method('write')
-            ->with($expected);
+            ->will(
+                $this->returnCallback(
+                    function ($data) use (&$buffer) {
+                        $buffer .= $data;
+                    }
+                )
+            );
 
         $response = new Response($conn);
         $response->writeHead();
+
+        $this->assertContains("HTTP/1.1 200 OK\r\n", $buffer);
+        $this->assertContains("X-Powered-By: React/alpha\r\n", $buffer);
+        $this->assertContains("Date: ", $buffer);
+        $this->assertContains("Transfer-Encoding: chunked\r\n", $buffer);
+        $this->assertContains("Connection: close\r\n\r\n", $buffer);
+        $this->assertContains("\r\n\r\n", $buffer);
```

Comment:
Are these changes needed? (See also below) They add quite a bit of noise to this PR.

My vote would be that the previous tests explicitly tested receiving a complete, valid HTTP response, while the new tests only test certain header fields. What is the motivation for this?